src/parse/app.rs

```diff
@@ -70,27 +70,30 @@ impl AppArgs {
                             ));
                         }
 
-                        if let Ok(lit) = input.parse::<LitBool>() {
-                            custom.insert(ident, CustomArg::Bool(lit.value));
-                        } else if let Ok(lit) = input.parse::<LitInt>() {
-                            if lit.suffix().is_empty() {
-                                custom.insert(
-                                    ident,
-                                    CustomArg::UInt(lit.base10_digits().to_string()),
-                                );
-                            } else {
-                                return Err(parse::Error::new(
-                                    ident.span(),
-                                    "argument has unexpected value",
-                                ));
-                            }
-                        } else if let Ok(p) = input.parse::<Path>() {
+                        // Parse as path
+                        if let Ok(p) = input.parse::<Path>() {
                             custom.insert(ident, CustomArg::Path(p));
                         } else {
-                            return Err(parse::Error::new(
-                                ident.span(),
-                                "argument has unexpected value",
-                            ));
+                            // Parse as literal
+                            match input.parse::<Lit>()? {
+                                Lit::Bool(lit) => {
+                                    custom.insert(ident, CustomArg::Bool(lit.value));
+                                }
+                                Lit::Int(lit) => {
+                                    if lit.suffix().is_empty() {
```

Comment:
if this comparison fails we should return the "argument has unexpected value" error as before, or even better we should return an "integer must be unsuffixed" error

Reply:
Cheers for the review! Squashed and pushed below